examples/vision/metric_learning.py

```diff
@@ -192,14 +192,17 @@ def train_step(self, data):
             # pairs, to be high. This loss will move embeddings for the
             # anchor/positive pairs together and move all other pairs apart.
             sparse_labels = keras.ops.arange(num_classes)
-            loss = self.compiled_loss(sparse_labels, similarities)
+            loss = self.compute_loss(x=None, y=sparse_labels, y_pred=similarities)
 
         # Calculate gradients and apply via optimizer.
         gradients = tape.gradient(loss, self.trainable_variables)
         self.optimizer.apply_gradients(zip(gradients, self.trainable_variables))
 
         # Update and return metrics (specifically the one for the loss value).
-        self.compiled_metrics.update_state(sparse_labels, similarities)
+        for metric in self.metrics:
+            # Calling `self.compile` will by default add a `keras.metrics.Mean` metric
+            metric.update_state(similarities)
```

Comment:
This will not generalize if there are more metrics passed to compile though -- prefer reusing the standard snippet from https://keras.io/keras_core/guides/custom_train_step_in_tensorflow/#a-first-simple-example

Reply:
Switched to standard snippet - at some later point we should circle back to how metrics are being dealt with because there's some weirdness in how we're handling Sum/Mean.